src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
+    if (openingComment) {
+      DOMChildrenOperations.removeDelimitedText(
```

Comment:
You shouldn't need to explicitly unmount the nodes in unmountComponent here; when we unmount a whole tree we only remove the root node, not every descendant individually. (DOMChildrenOperations would do the removal in the multi-child case.) See ReactDOMComponent.unmountComponent for comparison.